sway-core/src/semantic_analysis/ast_node/declaration.rs

```diff
@@ -206,11 +220,19 @@ impl TypedDeclaration {
     /// Retrieves the declaration as a struct declaration.
     ///
     /// Returns an error if `self` is not a `TypedStructDeclaration`.
-    pub(crate) fn expect_struct(&self) -> CompileResult<&TypedStructDeclaration> {
-        let warnings = vec![];
+    pub(crate) fn expect_struct(&self) -> CompileResult<TypedStructDeclaration> {
+        let mut warnings = vec![];
         let mut errors = vec![];
         match self {
-            TypedDeclaration::StructDeclaration(decl) => ok(decl, warnings, errors),
+            TypedDeclaration::StructDeclaration(decl_id) => {
+                let decl = check!(
+                    CompileResult::from(de_get_struct(decl_id.clone(), &self.span())),
```

Comment:
The span argument of `de_get_struct` is meant to represent the span at which the declaration is being accessed. So in this case, you should add an additional argument to `expect_struct` called `access_span: &Span` which is the span of the access.